api/src/main/java/org/openmrs/module/appointments/service/impl/SMSService.java

```diff
@@ -0,0 +1,116 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import static org.openmrs.module.appointments.util.DateUtil.convertUTCToGivenFormat;
+
+public class SMSService {
+
+    private final static String APPOINMTMENT_REMINDER_SMS_TEMPLATE = "sms.appointmentReminderSMSTemplate";
+    private final static String APPOINMTMENT_REMINDER_SMS_TEMPLATE_NO_PROVIDER = "sms.appointmentReminderSMSTemplateWithoutProvider";
+    private final static String SMS_TIMEZONE = "bahmni.sms.timezone";
+    private final static String SMS_DATEFORMAT = "bahmni.sms.dateformat";
+    private final static String SMS_URL = "bahmni.sms.url";
+    private static Logger logger = LogManager.getLogger(SMSService.class);
+    private OpenmrsLogin openmrsLogin;
+    private AppointmentVisitLocation appointmentVisitLocation;
+
+    public void setOpenmrsLogin(OpenmrsLogin openmrsLogin) {
+        this.openmrsLogin = openmrsLogin;
+    }
+
+    public void setAppointmentVisitLocation(AppointmentVisitLocation appointmentVisitLocation) {
+        this.appointmentVisitLocation = appointmentVisitLocation;
+    }
+
+    public String getAppointmentMessage(String name, String familyName, String id, Date appointmentDate, String service, List<String> providers, Location location) {
+        String smsTemplate = Context.getAdministrationService().getGlobalProperty(APPOINMTMENT_REMINDER_SMS_TEMPLATE);
+        String smsTemplateNoProvider = Context.getAdministrationService().getGlobalProperty(APPOINMTMENT_REMINDER_SMS_TEMPLATE_NO_PROVIDER);
+        String smsTimeZone = Context.getMessageSourceService().getMessage(SMS_TIMEZONE, null, new Locale("en"));
+        String smsDateFormat = Context.getMessageSourceService().getMessage(SMS_DATEFORMAT, null, new Locale("en"));
+        String date =convertUTCToGivenFormat(appointmentDate, smsDateFormat, smsTimeZone);
+        String helpdeskNumber = Context.getAdministrationService().getGlobalPropertyObject("clinic.helpDeskNumber").getPropertyValue();
+        LocationTag abc = Context.getLocationService().getLocationTagByName("Visit Location");
+        List<Location> locations = Context.getLocationService().getLocationsHavingAnyTag(Collections.singletonList(abc));
+        String clinicName = (abc != null && !locations.isEmpty()) ? locations.get(0).getName() : "xxxxx";
+        if (location != null) {
+            String cName = appointmentVisitLocation.getClinicName(location.getUuid());
+            if (StringUtils.isNotEmpty(cName)) {
+                clinicName = cName;
+            }
+        }
+        List<Object> arguments = new ArrayList<>();
+        arguments.add(name + " " + familyName);
+        arguments.add(id);
+        arguments.add(date);
+        arguments.add(org.springframework.util.StringUtils.collectionToCommaDelimitedString(providers));
+        arguments.add(service);
+        arguments.add(clinicName);
+        arguments.add(helpdeskNumber);
+
+        String message;
+        if (StringUtils.isBlank(smsTemplate)) {
```

Comment:
Consider converting this to a guard condition.. where if smsTemplate is Blank, you return and don't do anything after that. It doesn't need brackets.

Reply:
Gurpreet this condition was added as to check if the template in global property is empty or doesn't exist then it would evaluate the message using the template from message.properties file
Do we need to remove this condition and send empty message in that case
(which will remove's template from message.properties too)

Reply:
No. its okay. Let it be as-is then.